docs/contributing/running-tests.md

```diff
@@ -38,6 +38,23 @@ To run these tests locally, add `-PtestLatestDeps=true` to your existing `gradle
 
 Executing `./gradlew :instrumentation:<INSTRUMENTATION_NAME>:test --tests <GROOVY TEST FILE NAME>` will run only the selected test.
 
+### Ignoring warnings in tests
+
+During local development, you may want to ignore warnings in tests.
+
+To ignore warnings, formatting issues, or other non-fatal issues in tests, use
+
+```
+./gradlew test -Ddev=true -x spotlessCheck -x checkstyleMain
+```
+
+The `dev` flag
+
+- will ignore warnings in tests
+- will use a local build cache, which can speed up the build
```

Comment:
this part about speeding up the build is surprising to me, I'm checking out your branch to see if I can repro same

Reply:
I wasn't able to repro, they seem to run about the same (after the initial run which is always slow)

```
NORTHAMERICA+trstalna@Trask MINGW64 /c/git/opentelemetry-java-instrumentation (dev-mode)
$ ./gradlew spotlessApply -Ddev=true
Starting a Gradle Daemon (subsequent builds will be faster)
w: file:///C:/git/opentelemetry-java-instrumentation/settings.gradle.kts:87:9: 'setter for removeUnusedEntriesAfterDays: Int' is deprecated. Deprecated in Java

> Task :conventions:compileKotlin
w: file:///C:/git/opentelemetry-java-instrumentation/conventions/src/main/kotlin/otel.java-conventions.gradle.kts:297:5 'setter for dirMode: Int' is deprecated. Deprecated in Java
w: file:///C:/git/opentelemetry-java-instrumentation/conventions/src/main/kotlin/otel.java-conventions.gradle.kts:298:5 'setter for fileMode: Int' is deprecated. Deprecated in Java

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.8/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7m 21s
3087 actionable tasks: 2239 executed, 424 from cache, 424 up-to-date

NORTHAMERICA+trstalna@Trask MINGW64 /c/git/opentelemetry-java-instrumentation (dev-mode)
$ ./gradlew spotlessApply -Ddev=true

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.8/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 46s
3087 actionable tasks: 2 executed, 3085 up-to-date

NORTHAMERICA+trstalna@Trask MINGW64 /c/git/opentelemetry-java-instrumentation (dev-mode)
$ ./gradlew spotlessApply

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.8/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 44s
3087 actionable tasks: 2 executed, 3085 up-to-date

```

Reply:
ok - I'll take it out and collect more experience